onnxruntime/core/providers/cpu/math/cumsum.cc

```diff
@@ -48,12 +48,68 @@ void SumSlices(const Tensor& input, Tensor& output,
 }
 }  // namespace
 
+namespace cumsum_op {
+
+Status GetAxis(const Tensor* axis_tensor, const int64_t input_rank, int64_t& axis_out) {
+  if (!axis_tensor)
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Axis tensor must be provided to the CumSum op");
+
+  if (axis_tensor->Shape().NumDimensions() > 1)
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Axis tensor should be 0D or 1D");
+
+  if (axis_tensor->IsDataType<int32_t>()) {
+    axis_out = static_cast<int64_t>(axis_tensor->template Data<int32_t>()[0]);
+  } else if (axis_tensor->IsDataType<int64_t>()) {
+    axis_out = axis_tensor->template Data<int64_t>()[0];
+  } else {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Axis tensor should be of type `int32_t` or `int64_t`");
+  }
+
+  // validate input
+  if (axis_out < -input_rank || axis_out >= input_rank)
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Axis should be in the range [", -input_rank, ",", input_rank, ") but got: ", axis_out);
+
+  if (axis_out < 0)
+    axis_out = axis_out + input_rank;
```

Comment:
Could use HandleNegativeAxis() in providers/common.h

Reply:
Sure

Reply:
Fixed.